secboot/encrypt.go

```diff
@@ -74,3 +78,24 @@ func (key RecoveryKey) Save(filename string) error {
 	}
 	return osutil.AtomicWriteFile(filename, key[:], 0600, 0)
 }
+
+func RecoveryKeyFromFile(recoveryKeyFile string) (*RecoveryKey, error) {
+	f, err := os.Open(recoveryKeyFile)
+	if err != nil {
+		return nil, fmt.Errorf("cannot open recovery key: %v", err)
+	}
+	defer f.Close()
+	st, err := f.Stat()
+	if err != nil {
+		return nil, fmt.Errorf("cannot stat recovery key: %v", err)
+	}
+	if st.Size() != int64(len(RecoveryKey{})) {
+		return nil, fmt.Errorf("cannot read recovery key: unexpected size %v for the recovery key file", st.Size())
```

Comment:
I think all the other errors will include the filename in the err, but this one doesn't, it should probably do it on its own